File: 19 Remove Nth Node From End of List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
    struct ListNode* temp=head;
    struct ListNode* del;
    int count=0, pos;
    
    while(temp!=NULL)
    {
        ++count;
        temp=temp->next;
    }
    pos = count-n;
    if (pos == 0) {
        head = head->next;
        return head;
    }
    
    if (pos == 0) {
        head = head->next;
        return head;
    }
    
    temp = head;
    if (pos < 0)
    {
        return head;
    }
    for(count=0;count<pos-1;count++)
    {
        temp = temp->next;        
    }    
    if (temp->next != NULL) { 
        del = temp->next;
        temp->next = temp->next->next;
        free(del);
    }
    return head;
}



// Status: Accepted
// Runtime: 24 ms
// Memory Usage: 14.4 MB

File: 19 Remove Nth Node From End of List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {    
    dummy := &ListNode{Next: head}
    slow, fast := dummy, dummy
    
    for i := 0; i <= n; i++ {
        fast = fast.Next
    }
    
    for fast != nil {
        fast = fast.Next
        slow = slow.Next
    }
    
    slow.Next = slow.Next.Next
    
    return dummy.Next
}   
// Status: Accepted
// Runtime: 0 ms
// Memory Usage: 2.2 MB
